### PR TITLE
Add validation for additionalDisks

### PR DESCRIPTION
**Pull request by olamilekan000**
```
change validation to ensure that only Disk type values are allowed in the AdditionalDisks field.
```

Fixes [#3071](https://github.com/lima-vm/lima/issues/3071)

### CODE REVIEW

File: pkg/limayaml/validate.go [hunk 1]
```diff
@@ -136,6 +136,12 @@ func Validate(y *LimaYAML, warn bool) error {
 		return fmt.Errorf("field `memory` has an invalid value: %w", err)
 	}
 
+	for i, disk := range y.AdditionalDisks {
+		if disk.Name == "" {
+			return fmt.Errorf("additionalDisks[%d]: disk name is required", i)
```
Comment (alexandear) [inline pkg/limayaml/validate.go:141]: To be consistent with error messages in this function:
```suggestion
			return fmt.Errorf("field `additionalDisks[%d]` must be provided", i)
```

**Review by alexandear — APPROVED** [review]

File: pkg/limayaml/validate.go [hunk 2]
```diff
@@ -136,6 +136,12 @@ func Validate(y *LimaYAML, warn bool) error {
 		return fmt.Errorf("field `memory` has an invalid value: %w", err)
 	}
 
+	for i, disk := range y.AdditionalDisks {
+		if disk.Name == "" {
```
Comment (jandubois) [inline pkg/limayaml/validate.go:140]: This is not strict enough. `store.DiskDir` requires the name to be a valid identifier: https://github.com/lima-vm/lima/blob/2011af722e22d3c3663d0425768b38d760433b4d/pkg/store/store.go#L104-L105

For example:

```console
❯ l disk create "foo bar" --size 1gb
FATA[0000] identifier "foo bar" must match ^[A-Za-z0-9]+(?:[._-](?:[A-Za-z0-9]+))*$: invalid argument
```

So you could maybe just call `DiskDir` for validation:

```go
if _, err := store.DiskDir(disk.Name); err != nil {
    return fmt.Errorf("field `additionalDisks[%d].name is invalid`: %w, err)
}
```

**Review by jandubois — CHANGES_REQUESTED** [review]

File: pkg/limayaml/validate.go (hunk 2 above)
Comment (jandubois) [reply pkg/limayaml/validate.go:140]: Turns out you cannot import `store` into `limayaml` because that creates a circular import. This will change once the validation moves to `limatmpl`, but we are not there yet, so I guess you need to call `identifiers.Validate(disk.Name)` directly.

File: pkg/limayaml/validate_test.go [hunk 3]
```diff
@@ -48,6 +48,31 @@ func TestValidateProbes(t *testing.T) {
 	assert.Error(t, err, "field `probe[0].script` must start with a '#!' line")
 }
 
+func TestValidateAdditionalDisks(t *testing.T) {
+	images := `images: [{"location": "/"}]`
+
+	validDisks := `
+additionalDisks:
+  - name: "disk1"
+  - name: "disk2"
```
Comment (jandubois) [inline pkg/limayaml/validate_test.go:57]: Does not really matter, but our `.yamllint` style does not indent sequences. 

```suggestion
additionalDisks:
- name: disk1
- name: disk2
```

File: pkg/limayaml/validate_test.go [hunk 4]
```diff
@@ -48,6 +48,31 @@ func TestValidateProbes(t *testing.T) {
...
+
+	invalidDisks := `
+additionalDisks:
+  - name: ""
+`
+	y, err = Load([]byte(invalidDisks+"\n"+images), "lima.yaml")
+	assert.NilError(t, err)
+
+	err = Validate(y, false)
+	assert.Error(t, err, "field `additionalDisks[0].name is invalid`: identifier must not be empty: invalid argument")
```
Comment (jandubois) [inline pkg/limayaml/validate_test.go:73]: I think you need to update the error message. I would not go for an exact match, just check for the prefix.

**Review by jandubois — CHANGES_REQUESTED** [review]
I'm pretty sure the test needs to be updated to match.

You should also consider adding a test case for an invalid, but non-empty name, to verify it also throws an error.

**Review by jandubois — APPROVED** [review]
Sorry, I was wrong, the test is fine as-is.

Adding a non-empty invalid name might make sense, but otherwise this PR looks fine, thank you!
